src/main/java/sirius/kernel/commons/Json.java

```diff
@@ -267,17 +302,94 @@ public static ObjectNode getObject(ObjectNode objectNode, String fieldName) {
      * @param fieldName  the field name of the array to retrieve
-     * @return the array at the given field name
+     * @return the array at the given field name or an empty array if the field does not exist or is not an array
      */
     public static ArrayNode getArray(ObjectNode objectNode, String fieldName) {
-        return objectNode.withArray(JsonPointer.SEPARATOR + fieldName);
+        return tryGetArray(objectNode, fieldName).orElseGet(Json::createArray);
+    }
+
+    /**
+     * Tries to retrieve the {@link ArrayNode} at the given field name of the given {@link ObjectNode}.
+     *
+     * @param objectNode the node to retrieve the array from
+     * @param fieldName  the field name of the array to retrieve
+     * @return the array at the given field name or an empty optional if the field does not exist or is not an array
+     */
+    public static Optional<ArrayNode> tryGetArray(ObjectNode objectNode, String fieldName) {
+        JsonNode node = objectNode.get(fieldName);
+        if (node == null || !node.isArray()) {
+            return Optional.empty();
+        }
+        return Optional.of((ArrayNode) node);
+    }
+
+    /**
+     * Tries to retrieve the {@link JsonNode} at the given index of the given {@link ArrayNode}.
+     *
+     * @param arrayNode the node to retrieve the value from
+     * @param index     the index of the value to retrieve
+     * @return the value at the given index or an empty optional if the index is out of bounds or the node is null
+     */
+    public static Optional<JsonNode> tryGetAtIndex(ArrayNode arrayNode, int index) {
+        JsonNode node = arrayNode.get(index);
+        if (node == null || node.isNull()) {
+            return Optional.empty();
+        }
+        return Optional.of(node);
+    }
+
+    /**
+     * Tries to retrieve the {@link JsonNode} at the given field name of the given {@link ObjectNode}.
+     *
+     * @param objectNode the node to retrieve the value from
+     * @param fieldName  the field name of the value to retrieve
+     * @return the value at the given field name or an empty optional if the field does not exist or the node is null
+     */
+    public static Optional<JsonNode> tryGet(ObjectNode objectNode, String fieldName) {
+        JsonNode node = objectNode.get(fieldName);
+        if (node == null || node.isNull()) {
+            return Optional.empty();
+        }
+        return Optional.of(node);
+    }
+
+    /**
+     * Tries to retrieve the {@link JsonNode} at the given pointer of the given {@link JsonNode}.
+     *
+     * @param jsonNode the node to retrieve the value from
+     * @param pointer  the pointer to the value to retrieve
+     * @return the value at the given pointer or an empty optional if the pointer does not exist or the node is null
+     */
+    public static Optional<JsonNode> tryGetAt(JsonNode jsonNode, JsonPointer pointer) {
+        JsonNode node = jsonNode.at(pointer);
+        if (node == null || node.isNull()) {
```

Comment:
at doesn't return null but MissingNode
```suggestion
        if (node == null || node.isNull() || node.isMissingNode()) {
```